src/psyclone/tests/dynamo0p3_test.py

```diff
@@ -6635,12 +6653,14 @@ def test_no_halo_exchange_annex_dofs(tmpdir, monkeypatch,
                            api=TEST_API)
     psy = PSyFactory(TEST_API, distributed_memory=True).create(invoke_info)
     result = str(psy.gen)
+
     assert LFRicBuild(tmpdir).code_compiles(psy)
+
     if annexed:
         assert "CALL f1_proxy%halo_exchange" not in result
     else:
         assert "CALL f1_proxy%halo_exchange" in result
-    assert "CALL f2_proxy%halo_exchange" not in result
+        assert "CALL f2_proxy%halo_exchange" in result
```

Comment:
Please add a check in `if annexed` that the call does not exist in the output.

Reply:
Logical thing to do that I forgot :smiley: Done.